### PR TITLE
Civ 4667 dq enter expert details

### PR DESCRIPTION
**Pull request by GorilaNaranja**
### URL ###
/directions-questionnaire/expert-details

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-4667

### Change description ###
Defendant Response - DQ - Enter the expert's details

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    res.render(expertDetailsViewPath, {form});
+  } catch (error) {
+    next(error);
+  }
+});
+
+expertDetailsController.post(EXPERT_DETAILS_URL, async (req, res, next: express.NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const expertDetailsList: ExpertDetailsList = new ExpertDetailsList(req.body.items.map((expertDetail: ExpertDetails) => new ExpertDetails(
```
Comment (portizdev) [inline src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts:28]: Just an idea. This instruction could be included in a function in the Service and here simply make the call, the controller would be cleaner and we wouldn't have any logic here.

File: src/main/routes/urls.ts [hunk 2]
```diff
@@ -81,6 +81,8 @@ export const CITIZEN_WHY_DO_YOU_DISAGREE_FULL_REJECTION_URL = `${FULL_REJECTION_
 export const RESPONSE_INCOMPLETE_SUBMISSION_URL = `${BASE_CASE_RESPONSE_URL}/incomplete-submission`;
 export const CITIZEN_FR_AMOUNT_YOU_PAID_URL = `${FULL_REJECTION_URL}/how-much-have-you-paid`;
 export const DEFENDANT_SUMMARY_URL = `${DASHBOARD_URL}/:id/defendant`;
+export const EXPERT_DETAILS_URL = `${DIRECTIONS_QUESTIONNAIRE_URL}/expert-details`;
+export const EXPERT_EVIDENCE_URL = `${DIRECTIONS_QUESTIONNAIRE_URL}/expert-evidence`;
```
Comment (portizdev) [inline src/main/routes/urls.ts:85]: 'm pretty sure the URL name is wrong, this screen is already merged and I think the correct name is:
DQ_DEFENDANT_YOURSELF_EVIDENCE_URL

File: src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      expertDetail.firstName,
```
Comment (TatsyMaslova) [inline src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts:29]: we could do a constructor the way Kadir has done so we only pass req.body to it. how it is created would be hidden in the controller

File: src/main/views/features/directionsQuestionnaire/experts/expert-details.njk [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+{% from "govuk/components/error-message/macro.njk" import govukErrorMessage %}
+
+{% block pageTitle %}
+  {{ setPageTitle('PAGES.EXPERT_DETAILS.PAGE_TITLE') }}
+{% endblock %}
+
+{% set rowNumberToShowRemoveButton = 2 %}
+{% set length = form.model.items.length %}
+{% set isRemoveButtonHidden = form.model.items.length < rowNumberToShowRemoveButton %}
+{% set atLeastOneCourtOrderError = form.errorFor('items') %}
```
Comment (laperla11) [inline src/main/views/features/directionsQuestionnaire/experts/expert-details.njk:20]: You might need to change the name to your page

**Review by laperla11 — APPROVED** [review]

File: src/main/routes/urls.ts (hunk 2 above)
Comment (GorilaNaranja) [reply src/main/routes/urls.ts:85]: Done

**Review by laarmada — APPROVED** [review]

File: src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts (hunk 1 above)
Comment (GorilaNaranja) [reply src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts:28]: Done

File: src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts (hunk 3 above)
Comment (GorilaNaranja) [reply src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts:29]: Done

Comment (GorilaNaranja) [reply src/main/routes/features/directionsQuestionnaire/experts/expertDetailsController.ts]: Done

**Review by TatsyMaslova — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Comment by GorilaNaranja** [comment]
Pipeline issues

**Pull request closed without merging**